fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewPartitionFunctionChecker.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.analyzer;
+
+import com.google.common.collect.Maps;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.StringLiteral;
+import com.starrocks.catalog.FunctionSet;
+import com.starrocks.catalog.PrimitiveType;
+
+import java.util.Map;
+
+import static com.starrocks.sql.common.TimeUnitUtils.TIME_MAP;
+
+public class MaterializedViewPartitionFunctionChecker {
+    @FunctionalInterface
+    public interface CheckPartitionFunction {
```

Comment:
not a good name

Reply:
any good advise?  This PR only splits  the code into an independent file.